### PR TITLE
nistp521: add a comment to the P+P exceptional case in point_add.

### PR DESCRIPTION
**Pull request by agl**
This change adds a comment to the exceptional case in point_add that
handles the case of a doubling, which explains when this case may occur
during normal processing.

Thanks go to Antonio Sanso for noting this.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/ec/ecp_nistp521.c [hunk 1]
```diff
@@ -1253,6 +1253,20 @@ static void point_add(felem x3, felem y3, felem z3,
     /* ftmp5[i] < 2^61 */
 
     if (x_equal && y_equal && !z1_is_zero && !z2_is_zero) {
+        /* This is obviously not constant-time but it will almost-never happen
```
Comment (ekasper) [inline crypto/ec/ecp_nistp521.c]: There's already a comment on top of the file about this, perhaps either adjust it to say "almost never" and point to this comment; or remove it in favour of this one?

File: crypto/ec/ecp_nistp521.c [hunk 2]
```diff
@@ -1253,6 +1253,20 @@ static void point_add(felem x3, felem y3, felem z3,
...
+         * for ECDH / ECDSA. The case where it can happen is during scalar-mult
```
Comment (ekasper) [inline crypto/ec/ecp_nistp521.c]: Do we need to add the same comment to the other curves, too?

Comment (asanso) [reply crypto/ec/ecp_nistp521.c]: AFAIK the other curves are immune to this....

File: crypto/ec/ecp_nistp521.c (hunk 1 above)
Comment (richsalz) [reply crypto/ec/ecp_nistp521.c]: where's the other comment, in ```make_points_affine```?  I'll remove it or make that edit when I merge.

Comment (ekasper) [reply crypto/ec/ecp_nistp521.c]: Sorry, I meant "on top of the function". Above, on line 1159:

```
  * This function includes a branch for checking whether the two input points
  * are equal (while not equal to the point at infinity). This case never		  
  * happens during single point multiplication, so there is no timing leak for		  
  * ECDH or ECDSA signing. */
```

Comment (richsalz) [reply crypto/ec/ecp_nistp521.c]: I replaced this "This case never..." to end of comment with a "see comment below"

**Comment by richsalz** [comment]
merged to master.  thanks folks!

**Pull request closed without merging**